app/src/main/java/io/github/drp08/studypal/presentation/screens/ProfileScreen.kt

```diff
@@ -301,34 +310,299 @@ data object ProfileScreen : Screen {
+                                        fontWeight = FontWeight.SemiBold
+                                    )
+                                    Spacer(modifier = Modifier.width(2.dp))
+                                    Box(
+                                        contentAlignment = Alignment.CenterStart,
+                                        modifier = Modifier
+                                            .size(45.dp, 32.dp)
+                                            .clip(RoundedCornerShape(2.dp))
+                                            .border(
+                                                width = 1.dp,
+                                                color = Color.LightGray,
+                                                shape = RoundedCornerShape(5.dp)
+                                            )
+                                            .clickable { expandedShort.value = !expandedShort.value }
+                                    ) {
+                                        Text(
+                                            text = shortBreakTime.toString(),
+                                            modifier = Modifier.padding(start = 4.dp)
+                                        )
+                                        Spacer(modifier = Modifier.width(2.dp))
+                                        Icon(
+                                            Icons.Default.ArrowDropDown,
+                                            contentDescription = null,
+                                            Modifier
+                                                .align(Alignment.CenterEnd)
+                                                .fillMaxHeight()
+                                        )
+                                        DropdownMenu(
+                                            expanded = expandedShort.value,
+                                            onDismissRequest = { expandedShort.value = false }
+                                        ) {
+                                            timeToChoose.forEach { item ->
+                                                DropdownMenuItem(
+                                                    text = { Text(text = item.toString()) },
+                                                    onClick = {
+                                                        shortBreakTime =
+                                                            item //Todo: Connect to pomodoro timer
+                                                        expandedShort.value = false
+                                                    },
+                                                    modifier = Modifier.fillMaxSize()
+                                                )
+                                            }
+                                        }
+                                    }
+                                    Spacer(modifier = Modifier.width(2.dp))
```

Comment:
Can you try breaking all this down into smaller unit-identifiable functions? It feels way too big and clunky if one needs to edit this later

Reply:
no - i tried to write it like that originally - and there's a load of errors that crop up because of it - it's more time than its worth (at this stage in the project) - I would agree if they were checking our code but they're not

Reply:
Aight fineeee